### PR TITLE
ROX-19424: Adding pending approvals table

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds the basic table for Pending Approvals. I separated out the table cells into separate components so we may build the tables for the other tabs in a similar fashion (leaving out any table cells or adding any additional table cells, depending on which table we're looking at)

Screenshot 2023-10-23 at 7 50 37 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:1]: I was debating whether to separate each component into a different file but left it for now. If you think readability would be better as separate files, I can change that.

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:1]: Actually, for the sake of testing, maybe it would be better to separate them so I can make separate test files for each

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:51]: I think we want to handle this differently depending on what tab the user is viewing.

IMO:

- If there is no deferral update, simply show `Deferred (<original duration>)`
- If there is an unapproved deferral update:
-- Show the requested action as `Deferral pending update (<update duration>)` when on the Pending tab
-- Show the requested action as `Deferred (<original duration>)` when on the approved tab (this deviates from what we currently show in Risk Acceptance)

The thought here is that when viewing the Approved tab, the table will show what action the request was approved for, while still showing what action is pending on the Pending tab.

@mansursyed What do you think about the text in the Approved/Pending tabs when a request was approved, but has an unapproved update?

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:96]: Same thought here. I think if there is an unapproved update, the "Approved" tab should show the expiry date of only the approved portion of the request.

File: ui/apps/platform/src/services/VulnerabilityExceptionService.ts
Comment (dvail) [inline ui/apps/platform/src/services/VulnerabilityExceptionService.ts:73]: 👍

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:1]: I think this looks good as a single file, but separate would be OK too. The only components that look interesting to test might be the expiry-based ones.

**Review by dvail — APPROVED** [review]
This looks great and was very easy to review.

A couple of small comments, but otherwise good to go. The only thing I think we might want to change is how expiry is handled across the different tabs. I'll approve so that it doesn't hold this PR up but we should make a note to double check how we want to handle that in a follow up.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:25]: Ah yes just realized after rebasing 😬 . I'll reuse those

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:33]: Yea, that isn't necessary here. Will remove

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:51]: If you approve a deferral, and then update it, I thought it would go back to the `Pending approvals` tab. The second sub-bullet in the second bullet makes it seem like it would stay in the `Approved deferrals` tab.

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:51]: Oh actually, I'm remembering things again. So we actually show the updated deferral in both tabs, don't we? That's probably what you meant

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:51]: Okay so these are the states based on what you mentioned above:

- Pending approvals tab
    - `targetState = DEFERRAL & exceptionStatus = PENDING` -> Use `deferralReq`
    - `targetState = DEFERRAL & exceptionStatus = APPROVED_PENDING_UPDATE` -> Use `deferralUpdate`
- Approved Deferrals
    - `targetState = DEFERRAL & exceptionStatus = APPROVED` -> Use `deferralReq`
    - `targetState = DEFERRAL & exceptionStatus = APPROVED_PENDING_UPDATE` -> Use `deferralReq`

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:51]: We also now have a `Denied requests` tab. So I'm wondering what we should be showing for that one. If we defer a CVE for 30 days, approve it, update it to 90 days, and then deny it, what happens?

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx]: So I was thinking about this more, and was wondering about the following case:

* Defer a CVE for 7 days
* Approve request
* Update deferral to 30 days
* Approve request
* Update deferral to 90 days

What would you expect the expiry date for this request to be within the `Approved deferrals` tab? I'd probably expect it to be `30 days`. But what would the expiry be within `deferralReq` and `deferralUpdate` in this case? Would `deferralReq` show the original `7 days` or the latest approved `30 days`? (cc: @md2119 )

**Pull request merged**
